qoffer-domain/src/main/groovy/life/qbic/portal/portlet/offers/create/CreateOffer.groovy

```diff
@@ -8,5 +17,78 @@ package life.qbic.portal.portlet.offers.create
  * @since: 1.0.0
  * @author: Tobias Koch
  */
-class CreateOffer {
+class CreateOffer implements CreateOfferInput{
+
+    private CreateOfferDataSource dataSource
+    private CreateOfferOutput output
+
+    CreateOffer(CreateOfferDataSource dataSource, CreateOfferOutput output){
+        this.dataSource = dataSource
+        this.output = output
+    }
+
+    @Override
+    void createOffer(String tomatoId) {
+
+    }
+
+    @Override
+    void createNewOffer(String projectTitle, String projectDescription, Person customer, Person projectManager, List<ProductItem> productItems) {
+        QuotationId identifier = generateQuotationID(customer)
+
+        Quotation quotation = new Quotation(new Date(),null,customer,projectManager,projectTitle,projectDescription,productItems,calculateOfferPrice(productItems),identifier)
+        dataSource.saveOffer(quotation)
+    }
+
+    /**
+     * Method to generate the identifier of an offer with the project conserved part, the random part and the version
+     * @param customer which is required for the project conserved part
+     * @return
+     */
+    private static QuotationId generateQuotationID(Person customer){//do we want to have a person here? todo update the datamodellib
+        String projectConservedPart = customer.lastName.toLowerCase()
+        String randomPart = "abcd"
+        int version = 1
+
+        return new QuotationId(projectConservedPart,randomPart,version)
+    }
+
+    /**
+     * Method to calculate the price form the offer items
+     * @param items
+     * @return
+     */
+    private static double calculateOfferPrice(List<ProductItem> items, AffiliationCategory affiliationCategory){
+        //1. sum up item price
+        double offerPrice = 0
+        items.each {item ->
+            offerPrice += item.computeTotalCosts()
+        }
+        //2. add discount if available
+        double discount = getDiscount(affiliationCategory)
+        offerPrice *= discount
```

Comment:
I remember there to be a discussion on how to call this. As far as I remember the result was that it is not a discount but rather overheads added. So the logic is different
```suggestion
        //2. add overheads if applicable
        double overhead = getOverhead(affiliationCategory)
        offerPrice = offerPrice + offerPrice * overhead
```